src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IForLoopStatement.cs

```diff
@@ -0,0 +1,2041 @@
+        Right: IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32)
+            Left: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+            Right: ILiteralExpression (Text: 2) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2)
+    IExpressionStatement (OperationKind.ExpressionStatement)
+      IAssignmentExpression (OperationKind.AssignmentExpression, Type: System.Int32)
+        Left: ILocalReferenceExpression: j (OperationKind.LocalReferenceExpression, Type: System.Int32)
+        Right: IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32)
+            Left: ILocalReferenceExpression: j (OperationKind.LocalReferenceExpression, Type: System.Int32)
+            Right: ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IBlockStatement (0 statements) (OperationKind.BlockStatement)
+";
+            VerifyOperationTreeForTest<ForStatementSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17602, "https://github.com/dotnet/roslyn/issues/17602")]
+        public void IForLoopStatement_InitializerList()
+        {
+            string source = @"
+class C
+{
+    static void Main(string[] args)
+    {
+        int i = 1;
+        /*<bind>*/for (; i < 10; i = i + 1)
+        {
+        }/*</bind>*/
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IForLoopStatement (LoopKind.For) (OperationKind.LoopStatement)
+  Condition: IBinaryOperatorExpression (BinaryOperationKind.IntegerLessThan) (OperationKind.BinaryOperatorExpression, Type: System.Boolean)
+      Left: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+      Right: ILiteralExpression (Text: 10) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 10)
+  AtLoopBottom: IExpressionStatement (OperationKind.ExpressionStatement)
+      IAssignmentExpression (OperationKind.AssignmentExpression, Type: System.Int32)
+        Left: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+        Right: IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32)
+            Left: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+            Right: ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IBlockStatement (0 statements) (OperationKind.BlockStatement)
+";
+            VerifyOperationTreeForTest<ForStatementSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17602, "https://github.com/dotnet/roslyn/issues/17602")]
+        public void IForLoopStatement_DecreasingForIterator()
```

Comment:
@333fred most of this tests are ported over from existing tests  [http://index/?query=forloop&rightProject=Roslyn.Compilers.CSharp.Emit.UnitTests&file=CodeGen%5CForLoopsTests.cs&rightSymbol=zgzit1olwwc0](here) , hence the name and might be a repetition from ioperation perspective.